k9mail/src/main/java/com/fsck/k9/controller/ImapSyncInteractor.java

```diff
@@ -243,11 +226,30 @@ void syncRemoteDeletions(List<String> deletedMessageUids) throws IOException, Me
 
         if (moreMessages != null && moreMessages == MoreMessages.UNKNOWN) {
             final Date earliestDate = account.getEarliestPollDate();
-            int remoteStart = getRemoteStart(localFolder, imapFolder);
+            int remoteStart = syncHelper.getRemoteStart(localFolder, imapFolder);
             updateMoreMessages(earliestDate, remoteStart);
         }
     }
 
+    private void getAndOpenLocalFolder(SyncHelper syncHelper) throws MessagingException {
+        if (localFolder == null || !localFolder.isOpen()) {
```

Comment:
@Valodim The reason I made a method for that is to make the `syncRemoteDeletions` method viable for testing. If it runs before `performSync` , an NPE occurs since the folder instance variables have not been initialized yet, and after the `performSync` method runs, the folders will have been closed.
Is there a better way to handle this ? How about opening the folders in the constructor ?